department-of-veterans-affairs/vets-api
# Update service account seed for chatbot

## Summary

 This PR adds a new service account seed for the chatbot service. The previous PR was merged accidentally before the work was finalized, so this PR addresses some of the feedback contained therein.

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#65558


## Testing done
- Run the seeds script to add a new chatbot service account config; you can delete the previous one if it exists (has service_account_id = 'chatbot').
- Open the Rails console and generate an assertion.
```ruby
private_key = OpenSSL::PKey::RSA.new(
  File.read('spec/fixtures/sign_in/sample_service_account.pem')
)

payload = {
  'iss' => 'http://localhost:3001',
  'sub' => 'vets.gov.user+0@gmail.com',
  'aud' => 'http://127.0.0.1:3000/v0/sign_in/token',
  'iat' => Time.now.to_i,
  'exp' => 10.days.from_now.to_i,
  'scopes' => ['http://localhost:3000/v0/map_services/chatbot/token'],
  'service_account_id' => '88a6d94a3182fd63279ea5565f26bcb4',
  'jti' => SecureRandom.hex
}

JWT.encode(payload, private_key, 'RS256')
```
- Using the Sign in Service postman collection, obtain a service account token.
  - Paste the generated assertion as the value for the `signed_service_account_assertion` collection variable.
  - Send the `token` request (located in the `Service Account Auth` collection.)
- Verify the endpoint responds with a valid service account token.

## Screenshots
None.


## What areas of the site does it impact?

The chatbot service account config will be used as part of the effort to implement an endpoint that returns MAP STS tokens to service accounts.

## Acceptance criteria


## Requested Feedback
None.
